shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -253,9 +253,12 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite {
                   // ^ it may not end up being defined if there's an error
                 output(s"Defined " + td.kind.str + " " + td.nme.name)
                 val ttd = ctx.tyDefs(td.nme.name)
+                val thisTy = typer.TypeRef(ttd, ttd.tparams.flatMap(p => 
```

Comment:
Is there an easy way of avoiding too much logic duplication in `DiffTests`? It is very likely that code manipulating type provenance etc. will change in future Typer updates, and this would create a discrepancy. The logic should at least be moved and shared in a Typer method.